examples/cpp/modes/vtol/include/mode.hpp

```diff
@@ -0,0 +1,139 @@
+  { 
+    _vtol = std::make_shared<px4_ros2::VTOL>(*this);
+    _trajectory_setpoint = std::make_shared<px4_ros2::TrajectorySetpointType>(*this); 
+    _fixed_wing_setpoint = std::make_shared<px4_ros2::FwLateralLongitudinalSetpointType>(*this); 
+
+    _vehicle_global_position = std::make_shared<px4_ros2::OdometryGlobalPosition>(*this);
+  }
+
+  void onActivate() override 
+  {
+   VTOL::State initial_vtol_state = _vtol->get_current_state(); 
+
+    if(initial_vtol_state == VTOL::State::FIXED_WING){
+      _control_state = State::GoingNorth; 
+      std::cout << "VTOL Mode Activated. Flying North, climbing to 600m above sea level." << std::endl; 
+    } else if(initial_vtol_state == VTOL::State::UNDEFINED){
+      throw std::runtime_error("VTOL state undefined.");
+    } else{
+      _control_state = State::TransitionToFixedWing; 
+      std::cout << "VTOL Mode Activated. Transitioning to Fixed-Wing." << std::endl; 
+  }
+}
+
+  void onDeactivate() override {}
+
+  void updateSetpoint(float dt_s) override
+  {
+    switch (_control_state) {
+      
+      case State::TransitionToFixedWing: {
+
+        _vtol->to_fixedwing(); 
+
+        if (_vtol->get_current_state() == VTOL::State::TRANSITION_TO_FIXED_WING){
+
+          Eigen::Vector3f acceleration_sp = _vtol->compute_acceleration_setpoint_during_transition(); 
+          Eigen::Vector3f velocity_sp{NAN, NAN, 0.f}; 
+          float course_sp = 0.f; //align vehicle north 
+          float height_rate_sp = 0.f; 
+
+          _trajectory_setpoint->update(velocity_sp, acceleration_sp); 
+          _fixed_wing_setpoint->updateWithHeightRate(height_rate_sp, course_sp);
+
+        } else if(_vtol->get_current_state() == VTOL::State::FIXED_WING){
+
+          float altitude_sp = 600.f; 
+          float course_sp = 0.f; // due north 
+
+          _fixed_wing_setpoint->updateWithAltitude(altitude_sp, course_sp); 
+
+          _control_state = State::GoingNorth; 
+          std::cout << "Transition complete. Flying North, climbing to 600m above sea level." << std::endl; 
+        }
+      }
+      break; 
+
+      case State::GoingNorth: {
+
+        float altitude_sp = 600.f; 
+        float course_sp = 0.f; // due north 
+
+        _fixed_wing_setpoint->updateWithAltitude(altitude_sp, course_sp); 
+
+        if(_vehicle_global_position->position().z() >= (altitude_sp - 5.f)){
+          _control_state = State::TransitionToMulticopterAndHold;
+          std::cout << "Altitude Reached. Transitioning to Multicopter and holding." << std::endl;           
+        }
+      }
+      break; 
+
+      case State::TransitionToMulticopterAndHold: {
+
+        _vtol->to_multicopter(); 
+
+        if (_vtol->get_current_state() == VTOL::State::TRANSITION_TO_MULTICOPTER){
+
+          Eigen::Vector3f acceleration_sp = _vtol->compute_acceleration_setpoint_during_transition(); 
+          Eigen::Vector3f velocity_sp{NAN, NAN, 0.f}; 
+          float course_sp = M_PI; //align vehicle south
```

Comment:
You can use `float course_sp = 180_deg;`
And add `using namespace px4_ros2::literals;` at the top.